api/src/api/services/phases/handlers/squid-router-pay-phase-handler.ts

```diff
@@ -0,0 +1,257 @@
+import { BasePhaseHandler } from '../base-phase-handler';
+import RampState from '../../../../models/rampState.model';
+import logger from '../../../../config/logger';
+import { RampPhase } from 'shared';
+import { createPublicClient, encodeFunctionData, http } from 'viem';
+import { moonbeam } from 'viem/chains';
+import { getStatus } from '../../transactions/squidrouter/route';
+import { axelarGasService } from '../../../../contracts/AxelarGasService';
+import { privateKeyToAccount } from 'viem/accounts';
+import { MOONBEAM_FUNDING_PRIVATE_KEY } from '../../../../constants/constants';
+import { createMoonbeamClientsAndConfig } from '../../moonbeam/createServices';
+import { multiplyByPowerOfTen } from '../../pendulum/helpers';
+import Big from 'big.js';
+
+interface GpmFeeResult {
+  result: {
+    source_base_fee: number;
+    source_express_fee: {
+      express_gas_overhead_fee: number;
+      relayer_fee: number;
+      relayer_fee_usd: number;
+      express_gas_overhead_fee_usd: number;
+      total: number;
+      total_usd: number;
+    };
+    base_fee: number;
+    base_fee_usd: number;
+    execute_gas_multiplier: number;
+    execute_min_gas_price: string;
+    source_base_fee_string: string;
+    source_base_fee_usd: number;
+    destination_base_fee: number;
+    destination_base_fee_string: string;
+    destination_base_fee_usd: number;
+    source_confirm_fee: number;
+    destination_confirm_fee: number;
+    express_supported: boolean;
+    express_fee: number;
+    express_fee_string: string;
+    express_fee_usd: number;
+    express_execute_gas_multiplier: number;
+  };
+}
+interface AxelarScanStatusResponse {
+  is_insufficient_fee: boolean;
+  status: string; // executed or express_executed (for complete).
+  fees: {
+    base_fee: number; // in units of the native token.
+    source_base_fee: number;
+    destination_base_fee: number;
+    source_express_fee: {
+      total: number;
+    };
+    destination_express_fee: {
+      total: number;
+    };
+  };
+
+  id: string; // the id of the swap.
+}
+
+const AXELAR_POLLING_INTERVAL_MS = 10000; // 10 seconds
+const AXL_GAS_SERVICE_MOONBEAM = '0x2d5d7d31F671F86C782533cc367F14109a082712';
```

Comment:
This is the address of the contract that we need to call to fill up the gas I assume?